### PR TITLE
Numpy logspace

### PR DESCRIPTION
**Pull request by gyshi**
## Description ##
https://docs.scipy.org/doc/numpy/reference/generated/numpy.logspace.html

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by roywei** [comment]
@mxnet-label-bot add [numpy]

File: python/mxnet/ndarray/numpy/_op.py [hunk 1]
```diff
@@ -1691,3 +1692,101 @@ def arcsin(x, out=None, **kwargs):
...
+             endpoint may or may not be included.
+    linspace : Similar to logspace, but with the samples uniformly distributed
+               in linear space, instead of log space.
+
+    Notes
+    -----
+    Logspace is equivalent to the code. Now wo only support axis = 0.
+
+    >>> y = np.linspace(start, stop, num=num, endpoint=endpoint)
+    ... # doctest: +SKIP
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: what's this?

File: python/mxnet/ndarray/numpy/_op.py [hunk 2]
```diff
@@ -1691,3 +1692,101 @@ def arcsin(x, out=None, **kwargs):
...
+    >>> plt.plot(x1.asnumpy(), y.asnumpy(), 'o')
+    [<matplotlib.lines.Line2D object at 0x...>]
+    >>> plt.plot(x2.asnumpy(), (y + 0.5).asnumpy(), 'o')
+    [<matplotlib.lines.Line2D object at 0x...>]
+    >>> plt.ylim([-0.5, 1])
+    (-0.5, 1)
+    >>> plt.show()
+    """
+    if isinstance(start, (list, tuple, _np.ndarray, NDArray)) or \
+            isinstance(stop, (list, tuple, _np.ndarray, NDArray)):
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]:
```python
    if isinstance(start, (list, tuple, _np.ndarray, NDArray)) or \
       isinstance(stop, (list, tuple, _np.ndarray, NDArray)):
```

File: tests/python/unittest/test_numpy_op.py [hunk 3]
```diff
@@ -1162,6 +1162,70 @@ def test_np_broadcast_arrays():
...
+            self._start = start
+            self._stop = stop
+            self._num = num
+            self._endpoint = endpoint
+            self._base = base
+            self._dtype = dtype
+
+        def hybrid_forward(self, F, x):
+            return x + F.np.logspace(self._start, self._stop, self._num, \
+                                         self._endpoint, self._base, self._dtype)
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]:
```python
            return x + F.np.logspace(self._start, self._stop, self._num,
                                     self._endpoint, self._base, self._dtype)
```

File: python/mxnet/ndarray/numpy/_op.py (hunk 2 above)
Comment (gyshi) [reply python/mxnet/ndarray/numpy/_op.py]: resolved

File: tests/python/unittest/test_numpy_op.py (hunk 3 above)
Comment (gyshi) [reply tests/python/unittest/test_numpy_op.py]: resolved

**Pull request closed without merging**